src/ccaligner.cpp

```diff
@@ -40,7 +40,7 @@ void printFooter()
     std::cout<<"https://github.com/saurabhshri/CCAligner/issues\n";
 }
 
-CCAligner::CCAligner(Params * parameters)
+CCAligner::CCAligner(std::shared_ptr<Params> parameters)
```

Comment:
Why does `parameters` have to be a shared pointer? There is no need for `CCAligner` to take (shared) ownership of it. A raw pointer is totally fine here.